float64 rotate_0
float64 rotate_1
float64 rotate_2

float64 rotate_3
float64 rotate_4
float64 rotate_5